File: TravelApp/src/main/java/org/example/travelapp/model/Trip.java
package org.example.travelapp.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Data
@NoArgsConstructor
@Table(name = "trips")
public class Trip {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column
    private String destination;

    private String startDate;
    private String endDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnoreProperties("trips")
    private User user;
}

File: TravelApp/travel_frontend/src/app/models/trip.ts
export interface Trip {
  id: string;
  destination: string;
  startDate: string;
  endDate: string;
  latitude: number;
  longitude: number;
  hotelName?: string;
  hotelLatitude?: number;
  hotelLongitude?: number;
  user: {
    id: string;
    name: string;
    email: string;
  };
}
